From 9d4c5c3656c2669540642973d13a62063d50b228 Mon Sep 17 00:00:00 2001
From: Tiejun Chen <tiejun.chen@windriver.com>
Date: Tue, 3 Mar 2009 12:39:05 +0800
Subject: [PATCH] fsl_mpc8349e_mitx: Support Flash

fsl_mpc8349e_mitx: Improve dts file
	* Add flash device node based on memory map found in user manual

Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 arch/powerpc/boot/dts/mpc8349emitx.dts |   34 +++++++++++++++++++++++++++++++-
 1 files changed, 33 insertions(+), 1 deletions(-)

diff --git a/arch/powerpc/boot/dts/mpc8349emitx.dts b/arch/powerpc/boot/dts/mpc8349emitx.dts
index e851250..ceed03f 100644
--- a/arch/powerpc/boot/dts/mpc8349emitx.dts
+++ b/arch/powerpc/boot/dts/mpc8349emitx.dts
@@ -295,7 +295,39 @@
 		compatible = "fsl,mpc8349e-localbus",
 			     "fsl,pq2pro-localbus";
 		reg = <0xe0005000 0xd8>;
-		ranges = <0x3 0x0 0xf0000000 0x210>;
+		ranges = <0x0 0x0 0xfe000000 0x00800000
+			  0x3 0x0 0xf0000000 0x00000210>;
+
+		flash@0,0 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			compatible = "cfi-flash";
+			reg = <0x0 0x0 0x800000>;
+			bank-width = <2>;
+			device-width = <1>;
+
+			HRCW@0 {
+				reg = <0x0 0x20000>;
+				read-only;
+			};
+
+			dtb@20000 {
+				reg = <0x20000 0xe0000>;
+			};
+
+			kernel@100000 {
+				reg = <0x100000 0x300000>;
+			};
+
+			fs@400000 {
+				reg = <0x400000 0x300000>;
+			};
+
+			u-boot@700000 {
+				reg = <0x700000 0x100000>;
+				read-only;
+			};
+		};
 
 		pata@3,0 {
 			compatible = "fsl,mpc8349emitx-pata", "ata-generic";
-- 
1.6.0.4

